# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/extensions/webrequest/StreamFilterParent.cpp
# Commit: 3b150d676b31
# Full Hash: 3b150d676b3122bffa703813876d1fbc35e10d17
# Author: Randell Jesup <rjesup@mozilla.com>
# Date: 2024-03-19 16:41:28
# Regressor Bug: 1356686
# File Overlap Count: 1
# Description:
#   Bug 1356686 - Do decompression off main thread in content r=necko-reviewers,valentin,extension-reviewers,robwu
#   
#   Content processes will now always retarget delivery of OnDataAvailable for Http
#   channels off the main thread. Consumers that were previously redirecting
#   off-main thread are not affected and their retargeting will stick, but any
# ==============================================================================

diff -r 54754a50d992 -r 3b150d676b31 toolkit/components/extensions/webrequest/StreamFilterParent.cpp
--- a/toolkit/components/extensions/webrequest/StreamFilterParent.cpp	Mon Mar 18 20:04:00 2024 +0000
+++ b/toolkit/components/extensions/webrequest/StreamFilterParent.cpp	Mon Mar 18 20:04:01 2024 +0000
@@ -230,7 +230,7 @@
   if (trsl) {
     return trsl->CheckListenerChain();
   }
-  return NS_ERROR_FAILURE;
+  return NS_ERROR_NO_INTERFACE;
 }
 
 NS_IMETHODIMP